### PR TITLE
feat: structured data core endpoints for creating / updating databases and tables

### PR DESCRIPTION
**Pull request by fontanierh**
It is just plumbing. Not including any of the endpoints that insert / update rows
trying to not do everything in one PR

### CODE REVIEW

File: core/bin/dust_api.rs [hunk 1]
```diff
@@ -1641,6 +1641,250 @@ async fn data_sources_delete(
...
+                    })),
+                }),
+            ),
+        },
+    }
+}
+
+#[derive(serde::Deserialize)]
+struct DatabasesListQuery {
+    data_source_id: Option<String>,
```
Comment (spolu) [inline core/bin/dust_api.rs:1725]: I think this should be a separate get no? what's the purpose of this query to limit the list to on database?

File: core/bin/dust_api.rs [hunk 2]
```diff
@@ -1641,6 +1641,250 @@ async fn data_sources_delete(
...
+    }
+}
+
+#[derive(serde::Deserialize)]
+struct DatabasesTablesListQuery {
+    offset: usize,
+    limit: usize,
+}
+
+async fn databases_tables_list(
```
Comment (spolu) [inline core/bin/dust_api.rs:1850]: I think there is no world in which we will not return all the tables? I'd remove offset/limit all the way down and up

**Review by spolu — DISMISSED** [review]
LGTM!

Left two small suggestion feel free to merge at any time

File: core/bin/dust_api.rs (hunk 1 above)
Comment (fontanierh) [reply core/bin/dust_api.rs:1725]: Not sure I understand what you mean -- You can either get all databases in the project, or get all databases inside of a data source depending on if `data_source_id` is `None` or not. 
Which one of these do you feel we should remove (and why) ?

File: core/bin/dust_api.rs (hunk 2 above)
Comment (fontanierh) [reply core/bin/dust_api.rs:1850]: You're right, the offet/limit part is maybe not super helpful.

> all the way down and up

Comment (fontanierh) [reply core/bin/dust_api.rs:1850]: Do you mean at HTTP API level + at store level ?

Comment (fontanierh) [reply core/bin/dust_api.rs:1879]: I've removed in from the HTTP api (still optionally available in `store`)

File: core/bin/dust_api.rs (hunk 1 above)
Comment (fontanierh) [reply core/bin/dust_api.rs:1725]: > limit the list to on database?

It's to limit the list on one **data_source**, not one database

**Pull request merged**